### ISSUE TITLE
Wrong share price after trading 

### ISSUE THREAD

**Issue by jennazenk** [melonproject/protocol#358]
- I create a fund, invest 20 MLN in it.  I call fund.performCalculations, it returns (raw response): 

````
[ { [String: '20000000000000000000'] s: 1, e: 19, c: [ 200000 ] },
  { [String: '0'] s: 1, e: 0, c: [ 0 ] },
  { [String: '0'] s: 1, e: 0, c: [ 0 ] },
  { [String: '0'] s: 1, e: 0, c: [ 0 ] },
  { [String: '0'] s: 1, e: 0, c: [ 0 ] },
  { [String: '20000000000000000000'] s: 1, e: 19, c: [ 200000 ] },
  { [String: '1000000000000000000'] s: 1, e: 18, c: [ 10000 ] } ]
````

ie in an human readable format: 
````
GAV: 20, 
NAV: 20, 
Share Price: 1, 
totalSupply: 20
`````

- Then I buy some XRP and call fund.performCalculations, which returns (raw response): 

````
[ { [String: '1.1851149581495002147903473729e+28'] s: 1, e: 28, c: [ 1, 18511495814950, 2147903473729 ] },
  { [String: '0'] s: 1, e: 0, c: [ 0 ] },
  { [String: '0'] s: 1, e: 0, c: [ 0 ] },
  { [String: '0'] s: 1, e: 0, c: [ 0 ] },
  { [String: '0'] s: 1, e: 0, c: [ 0 ] },
  { [String: '1.1851149581495002147903473729e+28'] s: 1, e: 28, c: [ 1, 18511495814950, 2147903473729 ] },
  { [String: '5.92557479074750107395173686e+26'] s: 1, e: 26, c: [ 5925574790747, 50107395173686 ] } ]
`````

ie. in a human readable format:
````
GAV: 11851149581.495002147903473729
NAV: 11851149581.495002147903473729, 
Share Price: 592557479.074750107395173686, 
totalSupply: 20
````
XRP has 6 decimals. My assumption is that share price might go wrong after trading assets whose share price !== 18.

**Comment by jennazenk** [comment]
Confirmed, seems to happen with assets whose decimal is different than 18. Just tried with LTC (8 decimals), same issue.

**Comment by travs** [comment]
Fixed in 61de4707c9f4850e95e5901266e725dd967649d2 and then that was simplified in 03e68a1792fe7aff731ea050d1bd150c11a9c9c0
